test/lib/ufe/testRename.py

```diff
@@ -96,6 +96,49 @@ def assertStageAndPrimAccess(
         assertStageAndPrimAccess(mayaSegment, ball35PathStr, usdSegment)
 
     def testRename(self):
+        # open tree.ma scene in test-samples
+        mayaUtils.openTreeScene()
+
+        # clear selection to start off
+        cmds.select(clear=True)
+
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|world|Tree_usd|Tree_usdShape')
+        usdPathSegment = usdUtils.createUfePathSegment('/TreeBase')
+        treebasePath = ufe.Path([mayaPathSegment, usdPathSegment])
+        treebaseItem = ufe.Hierarchy.createItem(treebasePath)
+
+        ufe.GlobalSelection.get().append(treebaseItem)
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # set the edit target to the root layer
+        stage.SetEditTarget(stage.GetRootLayer())
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+        self.assertTrue(stage.GetRootLayer().GetPrimAtPath("/TreeBase"))
+
+        # get default prim
+        defualtPrim = stage.GetDefaultPrim()
+        self.assertEqual(defualtPrim.GetName(), 'TreeBase')
+
+        # TreeBase has two childern: leavesXform, trunk
+        assert len(defualtPrim.GetChildren()) == 2
+
+        # get prim spec for defualtPrim
+        primspec = stage.GetEditTarget().GetPrimSpecForScenePath(defualtPrim.GetPath());
+
+        # set primspec name
+        primspec.name = "TreeBase_potato"
+
+        assert stage.GetPrimAtPath('/TreeBase_potato')
+        assert stage.GetPrimAtPath('/TreeBase_potato/leavesXform')
+
+        # HS, 6-May-2020. defualtPrim in null?? This can't be null....
+        # defualtPrim = stage.GetDefaultPrim()
+        # self.assertEqual(defualtPrim.GetName(), 'TreeBase_potato')
```

Comment:
@ppt-adsk  Unless I am missing something really obvious, defualtPrim can not be null in line 138??

Both assert passed in line 134,135 which to me indicates the subtree under the renamed prim is in a good state.

Reply:
OK, I did a little more digging into this and the good news is that the tree/subtree  graph is in a good state after the rename.  It is just that we can't no longer rely on the stage which looks like a bug on USD side to me or perhaps there is a rational reason for this behavior that I don't know.
 
```
# set primspec name
primspec.name = "TreeBase_potato"

potatoPrim = stage.GetPrimAtPath('/TreeBase_potato')
self.assertEqual(potatoPrim.GetName(), 'TreeBase_potato')   // Pass
```

Traverse the stage

```
/TreeBase_potato
/TreeBase_potato/leavesXform
/TreeBase_potato/leavesXform/leaves
/TreeBase_potato/trunk
```